forge-game/src/main/java/forge/game/cost/CostDiscard.java

```diff
@@ -174,7 +174,7 @@ else if (type.equals("DifferentNames")) {
             }
             if (sameName) {
                 for (Card c : handList) {
-                    if (CardLists.filter(handList, CardPredicates.nameEquals(c.getName())).size() > 1) {
+                    if (CardLists.count(handList, CardPredicates.nameEquals(c.getName())) > 1) {
```

Comment:
i think there is `Iterables.any` or something like that, that would stop if a valid card is found

Reply:
Yea but we need > 1